### PR TITLE
impl: add GetJob client test

### PR DESCRIPTION
**Pull request by scotthart**
- copied over status matchers from google-cloud-cpp to avoid having to make the testing_util library public
- bazel dependency rules required creating a "common" library for both the testing_util target to depend on in addition to the others
- supporting bazel and cmake similarly resulted in creating a similar common cmake library
- added unit test for `Client::GetJob`

### CODE REVIEW

File: google/cloud/bigquery_unified/client_test.cc [hunk 1]
```diff
@@ -14,5 +14,47 @@
...
+using ::testing::Eq;
+using ::testing::Return;
+
+struct TestOption {
+  using Type = std::string;
+};
+
+TEST(BigQueryUnifiedClientTest, GetJobSuccess) {
+  auto mock_connection =
+      std::make_shared<google::cloud::bigquery_unified::MockConnection>();
```
Comment (dbolduc) [inline google/cloud/bigquery_unified/client_test.cc]: nit: just `MockConnection`

File: google/cloud/bigquery_unified/client_test.cc [hunk 2]
```diff
@@ -14,5 +14,47 @@
...
+
+  google::cloud::bigquery::v2::Job job;
+  job.mutable_job_reference()->set_project_id("my-project-id");
+  job.mutable_job_reference()->set_job_id("my-job-id");
```
Comment (dbolduc) [inline google/cloud/bigquery_unified/client_test.cc]: optional style nit:  I'd make this in the `EXPECT_CALL` lambda.

File: google/cloud/bigquery_unified/client_test.cc [hunk 3]
```diff
@@ -14,5 +14,47 @@
...
+      });
+
+  auto client =
+      Client(mock_connection, Options{}.set<TestOption>("client-test-option"));
+  google::cloud::bigquery::v2::GetJobRequest request;
+  request.set_project_id("my-project-id");
+  auto get_result = client.GetJob(request, call_options);
+  EXPECT_THAT(get_result, IsOk());
+  EXPECT_THAT(get_result->job_reference().project_id(), Eq("my-project-id"));
+  EXPECT_THAT(get_result->job_reference().job_id(), Eq("my-job-id"));
```
Comment (dbolduc) [inline google/cloud/bigquery_unified/client_test.cc]: Consider using `IsOkAndHolds(AllOf(ResultOf(...), ResultOf(...)))`

https://github.com/googleapis/google-cloud-cpp/blob/6c2eee90f6221ebe4836a1ecdbf242261e701ebe/google/cloud/bigquerycontrol/integration_tests/job_integration_test.cc#L117-L119

or the first one should be an `ASSERT_THAT(get_result, IsOk())`

**Review by dbolduc — APPROVED** [review]
I did not look too closely at the build scripts. I get why the version stuff needs to be pulled out.

Alternatively we could release the status matchers (by moving them into `google-cloud-cpp::mocks`). It is conceivable that applications are writing their interfaces in terms of our `google::cloud::Status`. Not saying we should, just that we could.

File: google/cloud/bigquery_unified/client_test.cc (hunk 1 above)
Comment (scotthart) [reply google/cloud/bigquery_unified/client_test.cc]: Done

File: google/cloud/bigquery_unified/client_test.cc (hunk 2 above)
Comment (scotthart) [reply google/cloud/bigquery_unified/client_test.cc]: Done

File: google/cloud/bigquery_unified/client_test.cc (hunk 3 above)
Comment (scotthart) [reply google/cloud/bigquery_unified/client_test.cc]: Thanks for reminding this mechanism exists. Done.

**Pull request merged**